numpy/core/src/npymath/npy_math_complex.c.src

```diff
@@ -445,28 +445,26 @@ npy_cpow@c@ (@ctype@ a, @ctype@ b)
     @type@ bi = npy_cimag@c@(b);
     @ctype@ r;
 
-    if (br == 0. && bi == 0.) {
-        return npy_cpack@c@(1., 0.);
-    }
-    if (ar == 0. && ai == 0.) {
-        if (br > 0 && bi == 0) {
-            return npy_cpack@c@(0., 0.);
+    /*Checking if in a^b, if b is zero.
+    If a is not zero then by definition of logarithm a^0 is one.
+    If a is also zero then as per IEEE 0^0 is best defined as 1.*/
+    if(br == 0. && bi == 0.){
+        return npy_cpack@c@(1.,0.);
+    }
+    /*Here it is already tested that for a^b, b is not zero.
+    So we are checking behaviour of a.*/
```

Comment:
I don't think thois comment is necessary?  It might make sense in the "generating an invalid" branch below.

Reply:
@seberg Could you please explain what do you mean by generating an invalid as I am not able to understand it

Reply:
I mean moving it down to the branch where you return NaN manually.  Maybe its a bit of a code layout thing in the diff, but I think that branch is the place where you may want to point out that `0^0` is already handled.

Reply:
All right got it. Thanks